csharp/extractor/Semmle.Extraction.CSharp.StubGenerator/StubGenerator.cs

```diff
@@ -56,13 +56,20 @@ private static void StubReference(ILogger logger, CSharpCompilation compilation,
         if (compilation.GetAssemblyOrModuleSymbol(reference) is not IAssemblySymbol assembly)
             return;
 
-        using var fileStream = new FileStream(FileUtils.NestPaths(logger, outputPath, path.Replace(".dll", ".cs")), FileMode.Create, FileAccess.Write);
-        using var writer = new StreamWriter(fileStream, new UTF8Encoding(false));
+        Func<StreamWriter> makeWriter = () =>
+        {
+            var fileStream = new FileStream(FileUtils.NestPaths(logger, outputPath, path.Replace(".dll", ".cs")), FileMode.Create, FileAccess.Write);
+            var writer = new StreamWriter(fileStream, new UTF8Encoding(false));
+            return writer;
+        };
+
+        using var visitor = new StubVisitor(assembly, makeWriter);
 
-        writer.WriteLine("// This file contains auto-generated code.");
-        writer.WriteLine($"// Generated from `{assembly.Identity}`.");
+        if (!assembly.Modules.Any(m => visitor.IsRelevantNamespace(m.GlobalNamespace)))
+            return;
 
-        var visitor = new StubVisitor(assembly, writer);
+        visitor.StubWriter.WriteLine("// This file contains auto-generated code.");
+        visitor.StubWriter.WriteLine($"// Generated from `{assembly.Identity}`.");
 
         visitor.StubAttributes(assembly.GetAttributes(), "assembly: ");
```

Comment:
I think we're not stubbing any assembly attributed at the moment, so this could be removed. Also, if we stubbed them, then the generated stubs would probably not compile due to each file containing assembly attributes.